test/e2e/e2e_test.go

```diff
@@ -219,22 +222,36 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
 				}, util.Timeout, util.Interval).Should(gomega.Equal(3))
 			})
 
-			ginkgo.By("Verify their position in ClusterQueue", func() {
-				gomega.Expect(info.Items[0].PositionInClusterQueue).To(gomega.BeNumerically("==", 0))
-				gomega.Expect(info.Items[1].PositionInClusterQueue).To(gomega.BeNumerically("==", 1))
-				gomega.Expect(info.Items[2].PositionInClusterQueue).To(gomega.BeNumerically("==", 2))
-			})
-
-			ginkgo.By("Verify two first workloads have high priority and 0 LocalQueue position", func() {
-				gomega.Expect(info.Items[0].Priority).To(gomega.Equal(highPriorityClass.Value))
-				gomega.Expect(info.Items[1].Priority).To(gomega.Equal(highPriorityClass.Value))
-				gomega.Expect(info.Items[0].PositionInLocalQueue).To(gomega.BeNumerically("==", 0))
-				gomega.Expect(info.Items[1].PositionInLocalQueue).To(gomega.BeNumerically("==", 0))
+			ginkgo.By("Verify their positions and priorities", func() {
+				expectedWls := []visibility.PendingWorkload{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: ns.Name,
+						},
+						Priority:               highPriorityClass.Value,
+						PositionInLocalQueue:   0,
+						PositionInClusterQueue: 0,
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: ns.Name,
+						},
+						Priority:               highPriorityClass.Value,
+						PositionInLocalQueue:   0,
+						PositionInClusterQueue: 1,
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: ns.Name,
+						},
+						Priority:               lowPriorityClass.Value,
+						PositionInLocalQueue:   1,
+						PositionInClusterQueue: 2,
+					},
+				}
+				gomega.Expect(info.Items).To(gomega.BeComparableTo(expectedWls, cmpopts.IgnoreFields(visibility.PendingWorkload{}, "Name", "LocalQueueName")))
```

Comment:
Why excluding the LocalQueueName here, in favor of the additional step? I think it would read easier to have all the properties here.

Reply:
The two first workloads have different names and localQueues, since we don't know their order I wanted to avoid checking this. An alternative would be to wait until one the first workloads gets created or add an additional priority class.

Reply:
I've added an additional priority class